### ISSUE TITLE
Solved?

### ISSUE THREAD

**Issue by oguzhanbahadir** [basit3407/testing-custom-sound-notifications#1]
Hi,
Did you solve this problem? Im facing same issue and cant figure it out.

**Comment by basit3407** [comment]
Yup, it was solved. You need to set the notification channel in android.

**Comment by oguzhanbahadir** [comment]
I already set the notification channel but still default voice incoming. Can you please share the code if you have a working one?

**Comment by basit3407** [comment]
Delete all the notification channels through code and then again set the notification channel, set the priority of this channel to maximum. And ok i will share code.

**Comment by basit3407** [comment]
**notificationChannels.ts**

import {
  AndroidImportance,
  setNotificationChannelAsync,
} from "expo-notifications";

import { Platform } from "react-native";

export const channelIdTimeBeginsWithAdhan = "timestartNotificationsWithAdhan";
export const channelIdTimeBeginsWithoutAdhan =
  "timestartNotificationWithoutAdhan";

export const channelIdReminder = "reminderNotification";

const importance = AndroidImportance.MAX;
export const vibrationPattern = [0, 250, 250, 250];
export const adhanSound = "adhan.wav";

if (Platform.OS === "android") {
  setNotificationChannelAsync(channelIdReminder, {
    name: "Reminder Notifications",
    importance,
    vibrationPattern,
  });

  setNotificationChannelAsync(channelIdTimeBeginsWithAdhan, {
    name: "Time Begins Notifications With Adhan",
    importance,
    vibrationPattern,
    sound: adhanSound,
  });

  setNotificationChannelAsync(channelIdTimeBeginsWithoutAdhan, {
    name: "Time Begins Notifications Without Adhan",
    vibrationPattern,
    importance,
  });
}


**scheduleReminderNotifications.ts**

/**
 * schdules reminder notification to remind that salah has not been performed at 45 minutes before the given next salah time
 * @param nextSalahTime time of next salah
 * @param salah name of salah
 */
export const scheduleReminderNotification = async (
  nextSalahTime: Date,
  salah: compulsoryPrayer
): Promise<void> => {
  if (Platform.OS === "ios") {
    const hasPushNotificationPermissionGranted =
      await allowsNotificationsAsync();

    if (!hasPushNotificationPermissionGranted) return;
  }

  const data: notificationData = {
    type: "reminder",
    key: salah,
  };

  const trigger: DateTriggerInput = {
    channelId: channelIdReminder,
    date: nextSalahTime.getTime() - 45 * 60 * 1000,
  };

  /**
   * Title that will be displayed in notification
   */
  const title = reminderTitleAddon + salah;

  await scheduleNotificationAsync({
    trigger,
    content: {
      data,
      title,
      vibrate: vibrationPattern,
    },
  });
};

export async function allowsNotificationsAsync(): Promise<boolean> {
  const settings = await requestPermissionsAsync({
    ios: {
      allowAlert: true,
      allowBadge: true,
      allowSound: true,
      allowAnnouncements: true,
    },
  });
  return (
    settings.granted ||
    settings.ios?.status === IosAuthorizationStatus.PROVISIONAL
  );
}



**app.config.js**

/* eslint-disable no-undef */
import "dotenv/config";

export default {
  name: "Islamic-Pray-Tracker",
  slug: "islamic-pray-tracker",
  icon: "./assets/icon.png",
  runtimeVersion: "1.1.4",
  version: "1.1.4",
  scheme: "islmprytrckr",
  extra: {
    // GOOGLE_API_KEY: process.env.GOOGLE_API_KEY,
    GOOGLE_WEB_CLIENT_ID: process.env.GOOGLE_WEB_CLIENT_ID,
  },
  android: {
    package: "com.bm.islamicpraytracker",
    icon: "./assets/icon.png",
    adaptiveIcon: {
      foregroundImage: "./assets/adaptive-icon.png",
    },
    versionCode: 15,
    googleServicesFile: "./google-services.json",
  },
  splash: {
    image: "./assets/splash.png",
    resizeMode: "cover",
  },
  plugins: [
    [
      "expo-notifications",
      {
        // icon: "./local/path/to/myNotificationIcon.png",
        color: "#37b899",
        sounds: [
          "./assets/adhan/adhan.wav",
          // "./local/path/to/myOtherSound.wav",
        ],
      },
    ],
  ],
  orientation: "portrait",
  updates: {
    fallbackToCacheTimeout: 0,
    url: `https://exp.host/@basit3407/islamic-pray-tracker/index.exp?release-channel=staging&runtimeVersion=1.1.4`,
  },
  assetBundlePatterns: ["**/*"],
  ios: {
    supportsTablet: false,
    bundleIdentifier: "com.islamicpraytracker.bm",
    icon: "./assets/appstore.png",
  },
  web: {
    favicon: "./assets/favicon.png",
  },
  packagerOpts: {
    sourceExts: ["js", "jsx", "svgx"],
    transformer: "node_modules/react-native-svg-transformer/index.js",
  },
};

**Comment by oguzhanbahadir** [comment]
Thank you so much for your reply and your code. I will try as soon as possible.
